src/System.Private.ServiceModel/tools/PRService/sync.ashx

```diff
@@ -0,0 +1,227 @@
+<%@ WebHandler Language="C#" Class="PullRequestHandler" %>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+// This script assumes that the repo is set up as follows: 
+
+//[remote "origin"]
+//	url = https://github.com/dotnet/wcf
+//	fetch = +refs/heads/*:refs/remotes/origin/*
+//	fetch = +refs/pull/*/head:refs/remotes/origin/pr/*      <-- allows us to run git checkout pr/{id} 
+
+public class PullRequestHandler : IHttpHandler
+{
+    private const string _gitExecutablePath = @"C:\Program Files (x86)\Git\bin\git.exe";
+    private const string _gitRepoPath = @"N:\git\wcf-test";
+
+    private const int _gitSanityExecutionTimeoutMilliseconds = 1000;
+    private const int _gitExecutionTimeoutMilliseconds = 60000;
```

Comment:
Just FYI, ASP.NET has default request timeout of 110s. I am still debating myself if we should impose another timeout or just let it run until the request is aborted. I am okay with current code. Just bring this up for your consideration.